### PR TITLE
fhfdjkshfjk

### PR DESCRIPTION
**Pull request by sedroche**

### CODE REVIEW

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 1]
```diff
@@ -215,7 +215,10 @@ module.exports = function(mongodb) {
       logger.info("Rollup options = " + JSON.stringify(options));
       collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
         var filteredErr = helpers.ignoreSomeErrs(err);
-        return callback(filteredErr, results);
+        collection.find({}, function(err, docs) {
+          console.log('@@@@@doooocs', docs);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [This function has 181 lines, which is greater than the 150 lines authorized. Split it into smaller functions.](https://github.com/feedhenry/fh-messaging/blob/6f0cd8bffd652c973d5495b6debf150dd2b8b53c/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
* MINOR [Function has a complexity of 42 which is greater than 20 authorized.](https://github.com/feedhenry/fh-messaging/blob/6f0cd8bffd652c973d5495b6debf150dd2b8b53c/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 2]
```diff
@@ -215,7 +215,12 @@ module.exports = function(mongodb) {
       logger.info("Rollup options = " + JSON.stringify(options));
       collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
         var filteredErr = helpers.ignoreSomeErrs(err);
-        return callback(filteredErr, results);
+        collection.find({}, function(err, cursor) {
+          cursor.toArray(function(err, docs) {
+            console.log('@@@@@doooocs', docs);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 43 which is greater than 20 authorized.](https://github.com/feedhenry/fh-messaging/blob/17ac0a9818f0f556611990f711181ed491a53958/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
* MINOR [This function has 183 lines, which is greater than the 150 lines authorized. Split it into smaller functions.](https://github.com/feedhenry/fh-messaging/blob/17ac0a9818f0f556611990f711181ed491a53958/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 3]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
+      //     cursor.toArray(function(err, docs) {
+      //       console.log('@@@@@mbaas_appprequestdest doooocs', docs);
+      //       return callback(filteredErr, results);
+      //     });
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 4]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
+      //     cursor.toArray(function(err, docs) {
+      //       console.log('@@@@@mbaas_appprequestdest doooocs', docs);
+      //       return callback(filteredErr, results);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 5]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
+      //     cursor.toArray(function(err, docs) {
+      //       console.log('@@@@@mbaas_appprequestdest doooocs', docs);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 6]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
+      //     cursor.toArray(function(err, docs) {
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 7]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+              });
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 8]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+                });
+              });
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 9]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+                  return callback(filteredErr, results);
+                });
+              });
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 10]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+                  console.log('@@@@@mbaas_appprequestdest doooocs', docs);
+                  return callback(filteredErr, results);
+                });
+              });
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 11]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+                cursor.toArray(function(err, docs) {
+                  console.log('@@@@@mbaas_appprequestdest doooocs', docs);
+                  return callback(filteredErr, results);
+                });
+              });
+            });
+          });
+        });
       });
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 12]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+        collection.find({}, function(err, cursor) {
+          cursor.toArray(function(err, docs) {
+            console.log('@@@@@appprequestdest doooocs', docs);
+            var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+            logger.info("Rollup options = " + JSON.stringify(options));
+            collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+              var filteredErr = helpers.ignoreSomeErrs(err);
+              collection.find({}, function(err, cursor) {
+                cursor.toArray(function(err, docs) {
+                  console.log('@@@@@mbaas_appprequestdest doooocs', docs);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 13]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
-
-      var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
-      logger.info("Rollup options = " + JSON.stringify(options));
-      collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
-        var filteredErr = helpers.ignoreSomeErrs(err);
-        return callback(filteredErr, results);
+      mongodb.collection('apprequestsdest', function(err, collection) {
+        collection.find({}, function(err, cursor) {
+          cursor.toArray(function(err, docs) {
+            console.log('@@@@@appprequestdest doooocs', docs);
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 14]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+      // var options = {query : q, out: {reduce: metric, db: dbName}, scope: {ts: from}};
+      // logger.info("Rollup options = " + JSON.stringify(options));
+      // collection.mapReduce(mReduceConf.map, mReduceConf.reduce, options, function(err,results) {
+      //   var filteredErr = helpers.ignoreSomeErrs(err);
+      //   collection.find({}, function(err, cursor) {
+      //     cursor.toArray(function(err, docs) {
+      //       console.log('@@@@@mbaas_appprequestdest doooocs', docs);
+      //       return callback(filteredErr, results);
+      //     });
+      //   });
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: fh-messaging/lib/services/mbaas_metrics.js [hunk 15]
```diff
@@ -210,13 +210,35 @@ module.exports = function(mongodb) {
...
+      // });
```
Comment (fh-qube) [inline fh-messaging/lib/services/mbaas_metrics.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by fh-qube** [comment]
SonarQube analysis reported 15 issues:
* MAJOR 13 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [This function has 200 lines, which is greater than the 150 lines authorized. Split it into smaller functions.](https://github.com/feedhenry/fh-messaging/blob/7f65902ac7c20dde93f7f147fd85dfdfba6f8b9f/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
* MINOR [Function has a complexity of 46 which is greater than 20 authorized.](https://github.com/feedhenry/fh-messaging/blob/7f65902ac7c20dde93f7f147fd85dfdfba6f8b9f/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by sedroche** [comment]
Retest this please

**Comment by fh-qube** [comment]
SonarQube analysis reported 15 issues:
* MAJOR 13 major
* MINOR 2 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [This function has 200 lines, which is greater than the 150 lines authorized. Split it into smaller functions.](https://github.com/feedhenry/fh-messaging/blob/7f65902ac7c20dde93f7f147fd85dfdfba6f8b9f/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
* MINOR [Function has a complexity of 46 which is greater than 20 authorized.](https://github.com/feedhenry/fh-messaging/blob/7f65902ac7c20dde93f7f147fd85dfdfba6f8b9f/fh-messaging/lib/services/mbaas_metrics.js#L75) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Pull request closed without merging**